meetings/2021-07/july-13.md

```diff
@@ -0,0 +1,512 @@
+
+GCL: Yeah. And they objects like having an option too right?
+
+CP: Apparently yes. 
+
+GCL: Okay, that's already. I don't understand that but I'm not going to block this proposal. Either way. I'm just very perplexed about it.
+
+JHD: I have a clarifying question there, Caridy. if I call another Realms object function and pass it string and then is the object I get back. Does it have the internal Slots of a string? 
+
+CP: Nope. Nope, you're getting a proxy. The proxy does not have the internal slot, right? 
+
+JHD: So, Okay, yeah, thank you. As Mark has pointed out. There have been some blocking objections by folks who are represented by people in this room, but who have not debated these objections directly in this room. It's my understanding, which is basically saying the capability that browsers and node users already have should not be something we continue. We permit in a new API for reasons. I don't understand that all the arguments the arguments I've heard are very Good compelling Arguments for what default should be. The default should be what Justin and amp needs, right? Which is like the safer thing where I default it like you can't escape the abstraction, right? Like that that's very compelling argument but none of those arguments do need preclude, the capability. And there are use cases that require the capability. I specifically had these cases where I need to get objects with you. Read the journal slots, including the approxi to it. It's not sufficient and for built-ins like that. I know about, I could probably write a wrapper that like, detects an incoming thing with slots and reconstructs the thing with slots coming out. but like, that's going to very difficult and very easy to get wrong. So I guess it's if I think it's like my personal feeling is that 90% of the people who want solve this problem in this room, and that's a ballpark number. Some will, have their use cases, men and will. And we'll have much lesser much reduced forms of motivation to advance things, and the people who have their who have created, this sort of Deadlock and forced our alternative. Forced to callable Realms. boundary callable boundary Realms as an alternative. Will not have ever needed to come into this room and argue their points directly. They'll they have only done it by proxy and that is very frustrating because It doesn't matter. Solely what the Use cases the champion group has the proposal as it entered stage to solve a set of use cases. The champion group may only cover a subset of those, but all of those use cases are what like met the stage Two criteria we expect. So for the solution into the language, we agree. Those are these cases We want to solve this approach is effectively magically creating a new proposal that's already at stage 2, somehow. And asking for it to go to stage 3 while simultaneously magically discarding the existing proposal and I'm saying that in the sense that like some of those use cases are no longer achievable
```

Comment:
```suggestion
JHD: So, Okay, yeah, thank you. As Mark has pointed out, there have been some blocking objections by folks who are represented by people in this room, but who have not debated these objections directly in this room. It's my understanding that this is basically saying the capability (that browsers and node users already have) should not be something we permit in a new API for "reasons". All the arguments I've heard are very good, compelling arguments for what the default should be. The default should be what Justin and AMP needs - which is the safer thing, where you can't escape the abstraction. That's a very compelling argument, but none of those arguments preclude the capability, and there are use cases that require the capability. I specifically had these cases where I need to get objects with internal slots. A Proxy to it is not sufficient for built-ins like that. I could probably write a wrapper that like, detects an incoming thing with slots and reconstructs the thing with slots coming out, but that's going to very difficult and very easy to get wrong. So I guess it's if I think it's like my personal feeling is that 90% of the people who want solve this problem in this room, and that's a ballpark number. Some will have their use cases met, and we'll have much lesser, much reduced forms of motivation to advance things, and the people who have created this sort of deadlock and forced callable boundary Realms as an alternative will not have ever needed to come into this room and argue their points directly. They'll they have only done it by proxy, and that is very frustrating because It doesn't solely matter what use cases the champion group has - the proposal as it entered stage 2 solved a set of use cases. The champion group may only cover a subset of those, but all of those use cases are what were met. The stage 2 criteria means we expect the solution to go into the language, we agree those are these cases we want to solve. This approach is effectively magically creating a new proposal that's already at stage 2, somehow - and asking for it to go to stage 3 while simultaneously magically discarding the existing proposal. I'm saying that in the sense that some of those use cases are no longer achievable.
```